crates/pallet-domains/src/tests.rs

```diff
@@ -929,6 +929,62 @@ fn generate_invalid_domain_extrinsic_root_fraud_proof<T: Config + pallet_timesta
     })
 }
 
+#[test]
+fn test_invalid_domain_block_hash_fraud_proof() {
+    let creator = 0u64;
+    let operator_id = 1u64;
+    let head_domain_number = 10;
+    let mut ext = new_test_ext_with_extensions();
+    ext.execute_with(|| {
+        let domain_id = register_genesis_domain(creator, vec![operator_id]);
+        extend_block_tree(domain_id, operator_id, head_domain_number + 1);
+        assert_eq!(
+            HeadReceiptNumber::<Test>::get(domain_id),
+            head_domain_number
+        );
+
+        let bad_receipt_at = 8;
+        let mut domain_block = get_block_tree_node_at::<Test>(domain_id, bad_receipt_at).unwrap();
+
+        let bad_receipt_hash = domain_block.execution_receipt.hash();
+        let (fraud_proof, root) = generate_invalid_domain_block_hash_fraud_proof::<Test>(
+            domain_id,
+            bad_receipt_hash,
+            Digest::default(),
+        );
+        domain_block.execution_receipt.final_state_root = root;
+        domain_block.execution_receipt.domain_block_hash = H256::random();
+        DomainBlocks::<Test>::insert(bad_receipt_hash, domain_block);
```

Comment:
Generally, the `bad_receipt_hash` should be calculated based on the modified ER, although not cause any wrong here.